crowdsourcehinter/static/js/src/crowdsourcehinter.js

```diff
@@ -0,0 +1,305 @@
+function CrowdsourceHinter(runtime, element, data){
+
+    var executeHinter = true;
+    var isShowingHintFeedback = false;
```

Comment:
You're better off doing this by inspection than keeping an explicit variable. For example, this could be a class on a div which determines what's visible. Then you ask the div about the class. 

The core thing you want to avoid are redundant representations. Those can go out-of-sync with subtle bugs, and lead to big problems. If you can have as much a stateless system as possible, it will be more robust.